ruff_lsp/server.py

```diff
@@ -679,99 +968,166 @@ async def resolve_code_action(params: CodeAction) -> CodeAction:
         f"{CodeActionKind.SourceOrganizeImports.value}.ruff",
     ):
         # Generate the "Ruff: Organize Imports" edit
-        results = await _fix_document_impl(document, only="I001")
-        params.edit = _create_workspace_edits(document, results)
+        params.edit = await _fix_document_impl(document, only="I001")
 
     elif settings["fixAll"] and params.kind in (
         CodeActionKind.SourceFixAll,
         f"{CodeActionKind.SourceFixAll.value}.ruff",
     ):
         # Generate the "Ruff: Fix All" edit.
-        results = await _fix_document_impl(document)
-        params.edit = _create_workspace_edits(document, results)
+        params.edit = await _fix_document_impl(document)
 
     return params
 
 
 @LSP_SERVER.command("ruff.applyAutofix")
-async def apply_autofix(arguments: tuple[TextDocument]):
+async def apply_autofix(ls: server.LanguageServer, arguments: tuple[TextDocument]):
     uri = arguments[0]["uri"]
-    text_document = LSP_SERVER.workspace.get_text_document(uri)
-    results = await _fix_document_impl(text_document)
-    LSP_SERVER.apply_edit(
-        _create_workspace_edits(text_document, results),
-        "Ruff: Fix all auto-fixable problems",
-    )
+    document = Document.from_uri(uri)
+    workspace_edit = await _fix_document_impl(document)
+    if workspace_edit is None:
+        return
```

Comment:
Is this a behavior change? Like does this action now _not_ show up in some cases?

Reply:
> Is this a behavior change? 

No, it's not.

> Like does this action now not show up in some cases?

No, this action is registered and always shown on the client side using the VSCode extension. When invoked through VSCode, the code executes this command registered in the LSP.